src/main/java/com/github/fge/jsonpatch/OmitOperation.java

```diff
@@ -0,0 +1,54 @@
+package com.github.fge.jsonpatch;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.MissingNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.fge.jackson.JsonNumEquals;
+import com.github.fge.jackson.jsonpointer.JsonPointer;
+import com.google.common.base.Equivalence;
+import com.google.common.collect.Iterables;
+
+/**
+ *
+ */
+public class OmitOperation extends PathValueOperation {
+
+    private static final Equivalence<JsonNode> EQUIVALENCE
+        = JsonNumEquals.getInstance();
+    @JsonCreator
+    public OmitOperation(@JsonProperty("path") final JsonPointer path,
+                         @JsonProperty("value") final JsonNode value)
+    {
+        super("omit", path, value);
+    }
+
+    @Override
+    public JsonNode apply(final JsonNode node)
+        throws JsonPatchException
+    {
+        if (path.isEmpty()) {
+            if (EQUIVALENCE.equivalent(node, value)) {
+                return MissingNode.getInstance();
```

Comment:
Yeah, empty path means root of the target JSON node.  So, in "remove" op it means remove the entire JSON node.  A similar idea can be applied here.  Didn't think of it that way, but it is indeed scary, although defined behavior.